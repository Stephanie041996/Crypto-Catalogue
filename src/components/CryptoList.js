import _, { initial } from 'lodash';
import "../components/App.css"
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {GetCryptoList} from '../actions/cryptoActions'
import {Link} from 'react-router-dom'
const CryptoList = (props) =>{
    const [search, setSearch] = useState("");
    const dispatch = useDispatch();
    const cryptoList = useSelector(state => state.CryptoList);
 React.useEffect( () => {
     FetchData()
 // eslint-disable-next-line react-hooks/exhaustive-deps
 },[]);

 const FetchData = () =>{
     dispatch(GetCryptoList())
 }



  const ShowData = () => {
    const filteredCoins = cryptoList.data.filter(coin =>
        coin.name.toLowerCase().includes(search.toLowerCase())
      );

      if( !_.isEmpty(cryptoList.data)){
        return (
        <div className={"list-wrapper"}>
          {filteredCoins.map(el =>{
          return (
            <div className={"list-item"}>
                <img className={"list-img"}src={el.image} alt=""/>
          <div className={"End"}>
          <p >{el.id}</p>
          <Link to={`/Crypto/${el.id}`}>View</Link>
          </div>
          </div>
            )
          })}
        </div>
      )
    }
    
  if( cryptoList.loading){
    return <p>loading</p>
}
if(cryptoList.errorMessage !== ""){
    return <p>{cryptoList.errorMessage}</p>
}
return <p>Unable to get data</p>

    };


    return (
        <div>
       
          


            <div className="search-section">
                <p>Search: </p>
                <input type="text" placeholder='search' onChange={e => setSearch(e.target.value)}/>
                

            </div>
            {ShowData()}
        </div>
    )
}

export default CryptoList