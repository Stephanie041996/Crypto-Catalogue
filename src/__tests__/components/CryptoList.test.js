import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CryptoList from '../../components/CryptoList';
import store from '../../Store';

afterEach(cleanup);

it('should render the list of cryptos', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <CryptoList />
      </BrowserRouter>
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <CryptoList />
      </Provider>,
    ),
  ).toMatchSnapshot();
});
