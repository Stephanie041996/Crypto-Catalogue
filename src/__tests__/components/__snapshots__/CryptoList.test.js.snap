// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the list of cryptos 1`] = `
<DocumentFragment>
  <div>
    <div
      class="search-section"
      style="background-image: url(/img/bitcoin.png); background-position: 30% 50%; background-size: 250px 250px; background-repeat: no-repeat;"
    >
      <input
        placeholder="search"
        type="text"
      />
    </div>
    <p>
      loading
    </p>
  </div>
</DocumentFragment>
`;
